packages/simcore-sdk/src/simcore_sdk/node_ports_v2/port.py

```diff
@@ -159,9 +160,25 @@ async def _set(self, new_value: ItemConcreteValue) -> None:
             # convert the concrete value to a data value
             converted_value: ItemConcreteValue = self._py_value_converter(new_value)
 
-            if isinstance(converted_value, Path):
-                if not converted_value.exists() or not converted_value.is_file():
+            def _raise_if_file_not_on_disk(path: Path) -> None:
+                if not path.exists() or not path.is_file():
+                    log.warning("Could not find a valid file %s", path)
                     raise InvalidItemTypeError(self.property_type, str(new_value))
+
+            if isinstance(converted_value, Path):
+                if converted_value.is_symlink():
+                    symlink_target_path = Path(os.readlink(converted_value))
+                    if not symlink_target_path.is_absolute():
+                        # in the case of a relative symlink
+                        # since the working directory is different form the
+                        # one in which the symlink is placed,
```

Comment:
how do you account for recursive links? i.e. a link to a folder within the ``workdir`` that you are uploading

Reply:
one symlinks to files are supported, function `_raise_if_file_not_on_disk` ensures the symlink points to a file on the disk.